### PR TITLE
docs: edit and update DI providers

### PR DESCRIPTION
**Pull request by kapunahelewong**
Copy edit and re-org of DI providers info.

Edits copy and removes tokens and treeshaking
sections to reduce content duplication and keep
info focused.

### CODE REVIEW

File: aio/content/guide/architecture-services.md [hunk 1]
```diff
@@ -93,7 +93,7 @@ or in the `@NgModule()` or `@Component()` metadata
    When you provide the service at the root level, Angular creates a single, shared instance of `HeroService`
    and injects it into any class that asks for it.
    Registering the provider in the `@Injectable()` metadata also allows Angular to optimize an app
-   by removing the service from the compiled app if it isn't used.
+   by removing the service from the compiled app if it isn't used, a process known as tree-shaking.
```
Comment (aikidave) [inline aio/content/guide/architecture-services.md:96]:
```suggestion
   by removing the service from the compiled app if it isn't used, a process known as *tree-shaking*.
```
I don't think we are consistent with this, but I think introducing a term by putting it in italics is a good idea.

File: aio/content/guide/dependency-injection-providers.md [hunk 2]
```diff
@@ -1,235 +1,211 @@
 # Dependency providers
 
-A dependency [provider](guide/glossary#provider) configures an injector
-with a [DI token](guide/glossary#di-token),
-which that injector uses to provide the concrete, runtime version of a dependency value.
-The injector relies on the provider configuration to create instances of the dependencies
-that it injects into components, directives, pipes, and other services.
+You can use dependency providers to get your services where you need them.
+By configuring providers, you can access dependencies in components, directives, pipes, and other services.
```
Comment (aikidave) [inline aio/content/guide/dependency-injection-providers.md:4]:
```suggestion
By configuring providers, you can make services available to the parts of your application that need them.
```

File: aio/content/guide/dependency-injection-providers.md [hunk 3]
```diff
@@ -1,235 +1,211 @@
...
 {@a class-provider}
 
-## Alternative class providers
+## Configuring the injector to use alternative class providers
 
-Different classes can provide the same service.
-For example, the following code tells the injector
-to return a `BetterLogger` instance when the component asks for a logger
-using the `Logger` token.
+To configure the injector to return a different class that provides the same service, you can use `useClass`.
```
Comment (aikidave) [inline aio/content/guide/dependency-injection-providers.md:49]:
```suggestion
To configure the injector to return a different class that provides the same service, you can use the `useClass` property.
```

File: aio/content/guide/dependency-injection-providers.md [hunk 4]
```diff
@@ -1,235 +1,211 @@
...
 
-### Class providers with dependencies
+### Configuring class providers with dependencies
+
+If the alternative class providers have their own dependencies, specify both providers in the `providers` metadata option of the parent module or component.
 
-Another class, `EvenBetterLogger`, might display the user name in the log message.
+<code-example path="dependency-injection/src/app/providers.component.ts" region="providers-5"></code-example>
+
+In this example, `EvenBetterLogger` display the user name in the log message.
```
Comment (aikidave) [inline aio/content/guide/dependency-injection-providers.md:63]:
```suggestion
In this example, `EvenBetterLogger` displays the user name in the log message.
```

File: aio/content/guide/dependency-injection-providers.md [hunk 5]
```diff
@@ -1,235 +1,211 @@
...
 
 {@a aliased-class-providers}
 
-### Aliased class providers
+### Aliasing class providers
 
-Suppose an old component depends upon the `OldLogger` class.
-`OldLogger` has the same interface as `NewLogger`, but for some reason
-you can't update the old component to use it.
+To alias a class provider, specify the alias and the class provider in the `providers` array with `useExisting`.
```
Comment (aikidave) [inline aio/content/guide/dependency-injection-providers.md:74]:
```suggestion
To alias a class provider, specify the alias and the class provider in the `providers` array with the `useExisting` property.
```

File: aio/content/guide/dependency-injection-providers.md [hunk 6]
```diff
@@ -1,235 +1,211 @@
...
-<code-example path="dependency-injection/src/app/providers.component.ts" region="provider-9-ctor-interface"></code-example>
+{@a value-provider}
 
-<div class="alert is-helpful">
+## Injecting an object
 
-This might seem strange if you're used to dependency injection in strongly typed languages where an interface is the preferred dependency lookup key.
-However, JavaScript, doesn't have interfaces, so when TypeScript is transpiled to JavaScript, the interface disappears.
-There is no interface type information left for Angular to find at runtime.
+To inject an object, configure the injector with the `useValue` option.
```
Comment (aikidave) [inline aio/content/guide/dependency-injection-providers.md:119]: Just a note: let's be consistent with property or option. Our API says `useValue` and `useClass` are properties, so maybe that's the right way to go?

Comment (kapunahelewong) [reply aio/content/guide/dependency-injection-providers.md:119]: Agreed. `option` appears frequently so I'll switch them as we go.

**Review by alan-agius4 — APPROVED** [review]
Reviewed for: docs-libraries

**Review by aikithoughts — APPROVED** [review]
LGTM!

Reviewed-for: global-docs-approvers

**Pull request closed without merging**